spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -67,6 +67,24 @@ public void setOutputChannelName(String outputChannelName) {
 		this.outputChannelName = outputChannelName;
 	}
 
+	/**
+	 * Return the output channel.
+	 * @return the channel.
+	 * @since 4.3
+	 */
+	@Override
+	public MessageChannel getOutputChannel() {
+		if (this.outputChannelName != null) {
+			synchronized (this) {
+				if (this.outputChannelName != null) {
+					this.outputChannel = getChannelResolver().resolveDestination(this.outputChannelName);
+					this.outputChannelName = null;
```

Comment:
Wow! How have we missed this for the first cut?
Seems for me this is a good addition to our Late Binding family :smile: 

Although you haven't reflected this in the commit comment :wink: